runtime/src/main/java/org/corfudb/util/FileWatcher.java

```diff
@@ -88,13 +97,19 @@ private void poll() {
     }
 
     private void reloadNewWatchService() {
+        if (isStopped.get()) {
+            log.info("Watch service is stopped. Skip reloading new watch service.");
+            return;
+        }
+
         try {
             if (watchService != null) {
                 watchService.close();
             }
             watchService = FileSystems.getDefault().newWatchService();
             Path path = file.toPath().getParent();
-            path.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+            path.register(watchService, StandardWatchEventKinds.ENTRY_CREATE,
+                    StandardWatchEventKinds.ENTRY_DELETE, StandardWatchEventKinds.ENTRY_MODIFY);
             isRegistered.set(true);
```

Comment:
Probably isRegistered can be set to false at the start of reloadNewWatchService() or maybe handle a Throwable?
In case there's a throwable encountered in this method, it will be consumed by runSansThrow() and L61 will have isRegistered set to true (which should be false).

Reply:
Done. I also removed it from `start()` in case it fails outside catch Throwable or `LambdaUtils.runSansThrow(this::poll);`